### PR TITLE
Preliminary doc on the generation level pileup infrastructure.

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: Generation/doc/AddingPileupToGeneration.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+# Adding Pileup in FCCSW in Generation
+
+Some of the FCC scenarios have a large number of pileup collisions in each event.
+A full simulation of the full event is very expensive, but some studies may nevertheless require the full information.
+For this purpose, FCCSW includes infrastructure to add pileup at generator level.
+
+
+By default, the particle sources in the Generator package of FCCSW can handle pileup, but are set to a default of no pileup.
+The standard way of adding pileup is to add a `Pileup` tool, which has a and configure the number of pileup. There are different implemenations of this tool in FCCSW, depending on the desired distribution (Constant or Poisson-distributed number of pileup events mixed in each event)
```
Comment (hegner) [inline Generation/doc/AddingPileupToGeneration.md]: "has a and configure"

File: Generation/doc/AddingPileupToGeneration.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+
+
+# Adding Pileup in FCCSW in Generation
```
Comment (hegner) [inline Generation/doc/AddingPileupToGeneration.md]: I'd rather create one file which lists all options for Pile up handling and where things can be mixed. And then refer to this file.

File: Generation/doc/AddingPileupToGeneration.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+For this purpose, FCCSW includes infrastructure to add pileup at generator level.
+
+
+By default, the particle sources in the Generator package of FCCSW can handle pileup, but are set to a default of no pileup.
+The standard way of adding pileup is to add a `Pileup` tool, which has a and configure the number of pileup. There are different implemenations of this tool in FCCSW, depending on the desired distribution (Constant or Poisson-distributed number of pileup events mixed in each event)
+
+
+## Pileup parameters
+
+As a convenient access to FCC-hh pileup parameters, FCCSW includes some small Python dictionaries with the relevant values in `Generation/python/FCCPileupScenarios.py` which can be used in configuration files by, p.ex.,
```
Comment (hegner) [inline Generation/doc/AddingPileupToGeneration.md]: You may drop phrases "as a convenient access...". We should explain things, not advertise ;-)

File: Generation/doc/AddingPileupToGeneration.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Consult the file `FCCPileupScenarios.py` for more information on values in the dictionaries.
+
+
+## Adding Minimum Bias events in Pythia
+
+Using the standard event generator `Pythia`, minimum bias events can be generated and added on the fly using the `PythiaInterface` algorithm in the Generation package. A pythia `.cmd` file for FCC-hh pileup is provided in the Generation package under `Generation/data/Pythia_minbias_pp_100TeV.cmd`.
+Example usage can be seen in `Examples/options/simple_pythia.py`. The vertices are smeared according to the parameters in the pythia configuration file.
+
+
+## Adding Pileup from HepMC file
```
Comment (hegner) [inline Generation/doc/AddingPileupToGeneration.md]: And how do I prepare such a file? Is there any help for that?

**Comment by vvolkl** [comment]
Thanks for the comments overall!

File: Generation/doc/AddingPileupToGeneration.md (hunk 2 above)
Comment (vvolkl) [inline Generation/doc/AddingPileupToGeneration.md]: So, something like `PileupInFCCSW.md`, and then link to this file? This should definitely be added, but can the Generation part stay like this?

File: Generation/doc/AddingPileupToGeneration.md [hunk 5]
```diff
@@ -29,5 +29,8 @@ Example usage can be seen in `Examples/options/simple_pythia.py`. The vertices a
 
 ## Adding Pileup from HepMC file
 
-Pileup particles can also be pre-generated and read from disc in HepMC format. The `HepMCReader`-algorithm can again be configured with a `PileupTool`, taking the average number of pileup events to be read and the name of the HepMC file as parameters. An example can be found in `Generation/options/pileup_hepmcreader.py`
+Pileup particles can also be pre-generated and read from disc in HepMC format. The `HepMCReader`-algorithm can again be configured with a `PileupTool`, taking the average number of pileup events to be read and the name of the HepMC file as parameters. An example can be found in `Generation/options/pileup_hepmcreader.py`. 
+
+To prepare Minimum Bias HepMC files, any suitable generator may be used. For Pythia8, consult example [main42](http://home.thep.lu.se/~torbjorn/pythia81html/SampleMainPrograms.html) how to do this. The fcc-physics repository contains also contains an example how to use Pythia8 together with the FCC event data model.
```
Comment (hegner) [inline Generation/doc/AddingPileupToGeneration.md]: "contains also contains"

**Pull request merged**